packages/wonder-blocks-modal/components/modal-launcher/index.js

```diff
@@ -0,0 +1,114 @@
+ * the `modal` prop.
+ */
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+import Color from "wonder-blocks-color";
+import {View} from "wonder-blocks-core";
+
+import ModalLauncherPortal from "./modal-launcher-portal.js";
+
+type Props = {
+    // The modal to render.
+    //
+    // The modal will be rendered inside of a container whose parent is
+    // document.body. This allows us to use ModalLauncher within menus and
+    // other components that clip their content. If the modal needs to close
+    // itself by some other means than tapping the backdrop or the default
+    // close button a render callback can be passed. The closeModal function
+    // provided to this callback can be called to close the modal.
+    modal: React.Node | (({closeModal: () => void}) => React.Node),
+
+    // Use the children-as-function pattern to pass a openModal function for
+    // use anywhere within children. This provides a lot of flexibility in
+    // terms of what actions may trigger the ModalLauncher to launch the modal.
+    children: ({openModal: () => void}) => React.Node,
+
+    // If the parent needs to be notified when the modal is closed use
+    // this prop.  Do not use `onClose` on the modals themselves as this
+    // is used by ModalNavigator to determine when to remove the modal
+    // the backdrop from the screen.
+    onClose?: () => void,
+};
+
+type State = {
+    // Whether the modal should currently be open.
+    opened: boolean,
+};
+
+export default class ModalLauncher extends React.Component<Props, State> {
+    state = {opened: false};
+
+    _openModal = () => {
+        this.setState({opened: true});
+    };
+
+    _closeModal = () => {
+        this.setState({opened: false}, () => {
+            this.props.onClose && this.props.onClose();
+        });
+    };
+
+    _renderModal() {
+        if (typeof this.props.modal === "function") {
+            return this.props.modal({
+                closeModal: this._closeModal,
+            });
+        } else {
+            return this.props.modal;
+        }
+    }
+
+    render() {
+        const renderedChildren = this.props.children({
+            openModal: this._openModal,
+        });
+
+        return (
+            <View>
+                {renderedChildren}
+                {this.state.opened && (
+                    <ModalLauncherPortal>
+                        <View style={styles.modalPositioner}>
+                            {this._renderModal()}
+                        </View>
+                    </ModalLauncherPortal>
+                )}
+            </View>
+        );
+    }
+}
+
+const styles = StyleSheet.create({
+    modalPositioner: {
+        position: "fixed",
+        left: 0,
+        top: 0,
+
+        boxSizing: "border-box",
+        width: "100%",
+        height: "100%",
+        paddingLeft: 32,
+        paddingRight: 32,
+        paddingTop: 72,
+        paddingBottom: 72,
```

Comment:
We should set a percentage for the width/height directly in the modals themselves.

Reply:
I'm still not sure about setting percentages directly in the modals, since we expect to embed them directly in layouts too, but I'll try updating this to have the correct dimensions! (Asking Tabitha some more questions in Zeplin to try to get more clarity around behavior.)

Reply:
The width/height can always be override by the by passing an Aphrodite style.  I think the most common use case will be using the modals with the ModalLauncher.  As long as they can be styled appropriately in the inline case it should be fine.

Reply:
Maybe we can use React's `context` object to know if a modal is inside a `ModalLauncherPortal` and then set things appropriately.

Reply:
I've changed my mind on this one. A call site might also just as easily want the "I'm previewing a modal" behavior, and want the default proportional sizing. Since I can't really predict the breadth of use cases here, I'm gonna do the simple way you suggested of inlining the rules into the modal layout component itself :) thanks for pushing on it!